Git submodules rps pr

ajpowelsnl opened the pull request:
The proposed new code addresses feedback from PR#1015 [https://github.com/kokkos/kokkos-kernels/pull/1015](url) 

Here, we have implemented RAJAPerf Suite (RPS)-based performance testing using git submodules, vs. the merging many files approach outlined in PR#1015.  

To use this new RPS perf testing implementation in Kokkos-Kernels, simply type this command:

git clone --recursive --branch git_submodules_rps_PR https://github.com/ajpowelsnl/kokkos-kernels.git

PERFORMING THE COMMAND ABOVE CORRECTLY IS CRITICAL.

This command brings RPS in as a tpl dependency, but without affecting the parent repo (i.e., kokkos-kernels).

After the cloning step mentioned above, you should see:

ls kokkos-kernels/tpls/rajaperf/
blt  CMakeLists.txt  LICENSE  NOTICE  README.md  RELEASE  scripts  src  tpl

Next, create and change into a build directory in kokkos-kernels, and from there, run the cmake configuration:

cmake \
-DKokkos_ROOT=$KOKKOS_INSTALL_DIR \
-DKokkosKernels_ENABLE_TESTS=ON \
-DKokkosKernels_ENABLE_TESTS_AND_PERFSUITE=ON \
-DCMAKE_BUILD_TYPE=Debug \
-DCMAKE_CXX_STANDARD=14 .. \

Perform a build:

make -j12

If the build is successful, you will find the executables in:

pwd
/ascldap/users/ajpowel/kokkos-kernels/build_dot
[ajpowel@kokkos-dev-2 build_dot]$ 
[ajpowel@kokkos-dev-2 build_dot]$ ls tpls/rajaperf/bin/
       
Finally, you can run the RPS-based executable

( please see PR#1015 for instructions on setting up your $KOKKOS_KERNELS_DATA directory.
FYI --
echo $KOKKOS_KERNELS_DATA
/ascldap/users/ajpowel/data-for-kernels/data )
 
[##############################################

 ./tracked_testing --input-data $KOKKOS_KERNELS_DATA

###############################################

Expected output upon execution of the command above:




Reading command line input...


Reading command line input...

Setting up suite based on input...


RAJA performance suite run summary....
--------------------------------------

How suite will be run:
	 # passes = 1
	 Kernel size factor = 1
	 Kernel rep factor = 1
	 Output files will be named ./RAJAPerf*

The following kernels and variants (when available) will be run:

Variants
--------
Base_Seq
Lambda_Seq
RAJA_Seq
Kokkos_Lambda
Kokkos_Functor

Kernels(iterations/rep , reps)
-----------------------------
BLAS_DOT  (100000 , 50)
Sparse_SPMV:uur/dataset_0 (100000 , 50)
Sparse_SPMV:uur/dataset_1 (100000 , 50)


Run warmup kernel...


Running specified kernels and variants...


Generate run report files...


DONE!!!....

####################################################################

If all goes well, in RAJAPerf-timing.csv you will have this data:

Mean Runtime Report (sec.)  ,  ,  ,  ,  ,
Kernel                     , Base_Seq , Lambda_Seq , RAJA_Seq , Kokkos_Lambda , Kokkos_Functor
BLAS_DOT                   ,  Not run ,    Not run ,  Not run ,      0.000918 ,        Not run
Sparse_SPMV:uur/dataset_0  ,  Not run ,    Not run ,  Not run ,      0.002213 ,        Not run
Sparse_SPMV:uur/dataset_1  ,  Not run ,    Not run ,  Not run ,      0.002209 ,        Not run
~                                                                                                                                               
~                                                                                                             

Note the data for "Kokkos_Lambda."

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           WARNING: NO REVIEWERS HAVE BEEN REQUESTED FOR THIS PULL REQUEST!

brian-kelley commented on perf_test/sparse/KokkosSparse_sptrsv.cpp:
```diff
@@ -67,7 +67,8 @@
 //#define INTERNAL_CUSPARSE
 
-#if defined( KOKKOS_ENABLE_CXX11_DISPATCH_LAMBDA ) && (!defined(KOKKOS_ENABLE_CUDA) || ( 8000 <= CUDA_VERSION ))
+#if defined(KOKKOS_ENABLE_CXX11_DISPATCH_LAMBDA) && \
```
Revert ths file

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES

ajpowelsnl commented on perf_test/blas/blas/KokkosBlas_dot_perf_test.hpp:
```diff
@@ -0,0 +1,86 @@
+  // m is vector length                            
+  int m           = 100000;         
+  int repeat      = 1;              
+  bool layoutLeft = true;
```
@brian-kelley - I did a significant refactor of this test to reflect the changes you requested.

ajpowelsnl commented on perf_test/blas/blas/KokkosBlas_dot_perf_test_rps.cpp:
```diff
@@ -0,0 +1,99 @@
+          return std::make_tuple(
+                          setup_test<Kokkos::DefaultExecutionSpace, Kokkos::DefaultExecutionSpace::array_layout>(m, repeat));
+          },
+        [&](const int iteration, const int runsize, auto& data) {
```
@brian-kelley -- Please see my comment above.

ajpowelsnl replied:
@brian-kelley -- whoopsie!  Sorry, but I think my first comment was somehow lost.  Yes, you are correct in this case, but our design intent here is to provide an easy-to-use performance test blueprint for Kokkos Kernels developers.  In the context of this design intent, would it be ok if I left the capture by value and capture by reference expressions here, and elsewhere?

brian-kelley commented on perf_test/blas/blas/CMakeLists.txt:
```diff
@@ -5,3 +5,15 @@ KOKKOSKERNELS_ADD_EXECUTABLE(
     KokkosBlas_perf_test
     SOURCES KokkosBlas_perf_test.cpp
   )
+
```
Amy, I thought you added this directory but it was already there, so don't worry about renaming it.

ajpowelsnl replied:
@brian-kelley - ha!  I already did it.  Your suggested changes do make it clearer, so, it's probably a good idea to keep it, no?

ajpowelsnl commented on perf_test/PerfTestUtilities.cpp:
```diff
@@ -0,0 +1,33 @@
+// Created by Poliakoff, David Zoeller on 4/27/21.
+//
+
+#ifndef KOKKOSKERNELS_PERFTESTUTILITIES_HPP
```
@brian-kelley - no, as you pointed out, it should be in the .hpp file

ajpowelsnl commented on perf_test/blas/blas/KokkosBlas_dot_perf_test.cpp:
```diff
@@ -0,0 +1,269 @@
+//DANGER DANGER DANGER!! POTENTIAL MERGE ISSUES 
+
+
+template <class ExecSpace, class Layout>
```
@brian-kelley -- I did a significant refactor, per your suggestion (it was a good idea, particularly towards the goal of making tests relatively simple to debug, and about testing one thing per test).

ajpowelsnl commented on perf_test/blas/blas/KokkosBlas_team_dot_perf_test.hpp:
```diff
@@ -0,0 +1,132 @@
+                       double result = KokkosBlas::Experimental::dot(team, testMatrices.x, testMatrices.y);
+                       });
+
+    ExecSpace().fence();
```
@brian-kelley - timer.seconds() occurs in a commented bloc in this file.  I will delete this comment, because it does confuse things.  Performance results are output to the "RAJAPerf-timing.csv" and other files (to you point about recording the data somewhere).

ajpowelsnl commented on perf_test/blas/blas/KokkosBlas_team_dot_perf_test_rps.cpp:
```diff
@@ -0,0 +1,125 @@
+kernel_base_vector.push_back(rajaperf::make_kernel_base(
+        "BLAS_TEAM_DOT ",
+        run_params,
+        [=](const int m, const int repeat) {
```
@brian-kelley -- I totally get where you're coming from, but would it be OK to leave this way (in the spirit of providing a blueprint that _can_ be used when there is something to capture?

ajpowelsnl commented on perf_test/blas/blas/KokkosBlas_team_dot_perf_test.cpp:
```diff
@@ -0,0 +1,300 @@
+  Kokkos::Timer timer;
+
+  // Live test of dot:
+  Kokkos::parallel_for("TeamDotDemoUsage",
```
@brian-kelley -- Brian, I know this is a bit confusing, but the "KokkosBlas_team_dot_perf_test.cpp" does not use the headers I have created; this perf test is modeled on those you wrote, and is in the "standard" KK perf test implementation style.  The "KokkosBlas_team_dot_perf_test.hpp" has all the declarations needed by "KokkosBlas_team_dot_perf_test_rps.cpp," the RPS version.  Would you like for me to change the header file name to better reflect which style (KK or RPS) of the test is relevant?

ajpowelsnl commented on perf_test/sparse/KokkosSparse_spmv.cpp:
```diff
@@ -79,258 +79,145 @@
 
+  spmv_additional_data data(test);
+
+  std::cout << "running CRS matrix single vec" << std::endl;
```
@brian-kelley -- I don't believe this is our change; I think it was brought in when I merged with develop last week.  I _believe_ these are changes from Luc; he added "spmv_additional_data" to accommodate AMD requirements.

lucbv replied:
Yep that looks like it, sorry for leaving these in the test.
If you want me to remove them separately I can do that, but if you remove them in your PR I won't complain either : )

ajpowelsnl replied:
@lucbv - I will remove this debug output, and push the update.

ajpowelsnl replied:
@brian-kelley , @lucbv -- Are we good to go on the removal of this debug output in this file?

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

ajpowelsnl commented:
@brian-kelley - I believe that I've addressed all of your comments.  Would you please let me know if we are ready to merge this PR into "develop"?  Please do let me know if I need to do anything further?

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES

brian-kelley reviewed (changes requested):
One last thing is that ``trtri`` is really a level-3 function (triangular inverse) so it belongs in blas/blas3, not blas/blas1. Still, thanks for renaming blas to blas1 - the organization is nice now.

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+          test_data.A_matrix = create_random_x_vector(A_matrix);
+          // rm line 113 if build is good
+          //test.B_matrix = create_random_x_vector(B_matrix);
+          test_data.B_matrix = create_random_x_vector(B_matrix);
```
@brian-kelley - I'm running "clang-format -i <file>', and I'll adopt the style you mention in your comment.

ajpowelsnl replied:
@brian-kelley - I removed this file.

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+#include <cmath>
+#include <unordered_map>
+#include <Kokkos_Core.hpp>
+#include <KokkosSparse_CrsMatrix.hpp>
```
> One last thing is that `trtri` is really a level-3 function (triangular inverse) so it belongs in blas/blas3, not blas/blas1. Still, thanks for renaming blas to blas1 - the organization is nice now.

Thanks for the heads up on this point.  I moved the file to blas3

ajpowelsnl replied:
@brian-kelley -- I removed the perf_test/KokkosDot_test.cpp file, and I take your point about larding a file with unnecessary headers.  As for trtri, I will move it to the blas3 directory in this branch, as well.

ajpowelsnl commented on perf_test/blas/blas2/KokkosBlas2_gemv_perf_test.cpp:
```diff
@@ -116,52 +116,95 @@ int parse_inputs (Params& params, int argc, char **argv){
   bool useSerial = !useOMP && !useCUDA;
 
+  // Logic for runtime with PThreads 
+    if (useThreads)
```
@brian-kelley -- I clang-formatted it, and the result looks to be in line with the team coding standards.

ajpowelsnl replied:
@brian-kelley -- Sounds good - it should be fixed now.

ajpowelsnl commented on perf_test/PerfTestUtilities.cpp:
```diff
@@ -0,0 +1,27 @@
+// Created by Poliakoff, David Zoeller on 4/27/21.
+//
+
+#include "KokkosKernels_default_types.hpp"
```
@brian-kelley -- ok, I'll comment out the other headers, and test a build before committing.

ajpowelsnl replied:
@brian-kelley -- Sounds good.  I chucked all the unnecessary headers.

ajpowelsnl commented on perf_test/KokkosKernelsTrackedTesting.cpp:
```diff
@@ -0,0 +1,74 @@
+    // argument parsing for setting input data at runtime
+   
+ 	std::string inputDataPath;
+		if (argc == 1) {
```
@brian-kelley -- I ran clang formatter on this file, but I'm still not certain if that tool fixed the indentation the way you want it.

ajpowelsnl replied:
@brian-kelley -- I formatted all files in perf_test and perf_test/blas/blas1 (using clang-format).

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+        Kokkos::fence();
+        timer.reset();
+        double result_1D = KokkosBlas::dot(test_data.A_matrix, test_data.B_matrix);
+        double elapsed = timer.seconds();
```
@brian-kelley - ok, I deleted it.

ajpowelsnl commented on perf_test/PerfTestUtilities.hpp:
```diff
@@ -0,0 +1,108 @@
+  if (d) {
+    while ((dir = readdir(d)) != NULL) {
+      std::string nname = std::string(dir->d_name);
+      if (nname.find(".") != std::string::npos) {
```
@brian-kelley -- hmmm.... I don't know.  I might need to walk through this with you.  I'll fix the other stuff before I do this one, and circle back around to this one , once the other things are done.

DavidPoliakoff replied:
This is one I wrote a lot of, and I'm _terrible_ with filesystems. If you two want to work on this, please do, this is just the level at which I write fs code

ajpowelsnl replied:
@DavidPoliakoff -- OK, sounds good.  I'll work it with @brian-kelley

ajpowelsnl replied:
@brian-kelley -- I think we worked this change together in my "feature/gemv rps test" branch, and that change will merge in with that branch.  Is it OK if we take this approach, i.e., bring the recommended changes in through the "feature/gemv rps test" branch?

ajpowelsnl commented on perf_test/blas/blas1/KokkosBlas_team_dot_perf_test.cpp:
```diff
@@ -0,0 +1,274 @@
+  // Live test of dot:
+  Kokkos::parallel_for("TeamDotDemoUsage",
+                       policy(1, Kokkos::AUTO),
+                       KOKKOS_LAMBDA (const member_type& team)
```
@brian-kelley -- OK, I might need to go over this re-write with you &/or @DavidPoliakoff.

ajpowelsnl replied:
@brian-kelley -- I ended up addressing this comment in the "feature / gemv rps test" branch.  Would it be OK if we merged your recommended changes into develop via that branch, vs. this one?

ajpowelsnl commented on perf_test/blas/blas1/KokkosBlas_team_dot_perf_test_rps.cpp:
```diff
@@ -0,0 +1,115 @@
+#endif // KOKKOSKERNELS_ENABLE_TESTS_AND_PERFSUITE
+
+
+// Recall -- testData is a tempated class, 
```
@brian-kelley -- I removed the "Recall..." comment, but isn't it a good practice to comment the # endif to make sure you close your ifndef?

brian-kelley replied:
Correct, I was thinking just:
```
// Recall -- testData is a tempated class,
// setup_test is a templated function
```
could be removed

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+/*
```
@brian-kelley -- I removed this file today.

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+#include <PerfTestUtilities.hpp>
+
+#ifdef HAVE_CUSPARSE
+#include <CuSparse_SPMV.hpp>
```
@brian-kelley -- I think removal of perf_test/KokkosDot_test.cpp takes care of "sparse" and "SPMV" extraneous stuff.

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include "KokkosBlas1_dot.hpp"
+#include "KokkosKernels_Utils.hpp"
+// in test_common
```
@brian-kelley -- I removed this file, so, we don't have to look at ugly comments anymore, at least not in this file.

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+//#include "KokkosKernels_TestUtils.hpp"
+
+
+  //FUNCTION THAT IS PART OF KK for generating test matrices
```
@brian-kelley -- This file was removed.

ajpowelsnl commented on perf_test/KokkosDot_test.cpp:
```diff
@@ -0,0 +1,128 @@
+          DotTestData test_data;
+          using matrix_type   = DotTestData::matrix_type;
+          test_data.A_matrix = create_random_x_vector(A_matrix);
+          // rm line 113 if build is good
```
@brian-kelley -- I removed this file.

ajpowelsnl commented on perf_test/PerfTestUtilities.cpp:
```diff
@@ -0,0 +1,27 @@
+   inputDataPath = path_to_data;	
+
+};
+ std::string get_input_data_path(){
```
@brian-kelley -- I clang formatted this file.  Hopefully, it's now up to team standards.

ajpowelsnl commented on perf_test/blas/blas1/KokkosBlas_dot_perf_test_rps.cpp:
```diff
@@ -0,0 +1,96 @@
+#include <PerfTestUtilities.hpp>
+#endif
+
+// Recall -- testData is a tempated class, 
```
@brian-kelley -- comments removed.

ajpowelsnl commented on perf_test/blas/blas1/KokkosBlas_dot_perf_test_rps.cpp:
```diff
@@ -0,0 +1,96 @@
+        test_list kernel_base_vector;
+
+
+kernel_base_vector.push_back(rajaperf::make_kernel_base(
```
@brian-kelley -- I ran clang-formatter on this, and other files.

ajpowelsnl commented on perf_test/blas/blas1/KokkosBlas_team_dot_perf_test_rps.cpp.junk:
```diff
@@ -0,0 +1,11 @@
+
```
@brian-kelley -- done!

ajpowelsnl commented on perf_test/blas/blas2/KokkosBlas_dot_perf_test.hpp.reference:
```diff
@@ -0,0 +1,86 @@
+
```
@brian-kelley -- Done!  I didn't mean to add them in the first place -- sorry about that!

ajpowelsnl commented on perf_test/blas/blas2/KokkosBlas_dot_perf_test_rps.cpp.reference:
```diff
@@ -0,0 +1,99 @@
+/*
```
@brian-kelley -- I removed these files.

ajpowelsnl commented on perf_test/blas/blas2/gemv_notes.txt:
```diff
@@ -0,0 +1,36 @@
+FROM -- KokkosBlas2_gemv_perf_test.cpp 
```
@brian-kelley -- I moved it.  Sorry for the detritus that ended up being committed.

ajpowelsnl commented on perf_test/dot_tracked_testing.hpp:
```diff
@@ -0,0 +1,70 @@
+/*
```
@brian-kelley -- cool.  I'll chuck it.

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

brian-kelley commented on perf_test/blas/blas1/KokkosBlas_team_dot_perf_test.hpp:
```diff
@@ -0,0 +1,107 @@
+                                                    bool layoutLeft,
+                                                    const int numberOfTeams);
+
+// Must have the full function body, as templated functions are recipes for
```
Remove notes

brian-kelley commented on perf_test/blas/blas1/KokkosBlas_team_dot_perf_test.hpp:
```diff
@@ -0,0 +1,107 @@
+  Kokkos::parallel_for(
+      "TeamDotUsage_RPS_Warm_Up",
+      policy(testMatrices.numberOfTeams, Kokkos::AUTO),
+      KOKKOS_LAMBDA(
```
Use the functor you wrote in KokkosBlas_team_dot_perf_test.cpp instead of lambda, so that this will compile for CUDA without ``--expt-extended-lambda``.

ajpowelsnl reviewed (commented):
@brian-kelley -- Good catch!  I corrected the spelling error.

ajpowelsnl commented:
Hey Brian,

Thanks so much for the feedback.

RE -- Use the functor you wrote in KokkosBlas_team_dot_perf_test.cpp instead of lambda, so that this will compile for CUDA without --expt-extended-lambda;

The functor version of the team dot test is actually in the feature / gemv PR (#1085).

I merged the git_submodules_rps_pr with that branch (feature / gemv) this morning.

Would having the new and improved team dot test (i.e., with functor, and not KOKKOS_LAMBDA)  in the feature / gemv PR be OK, or would you like for me to handle it differently?

On different, but related points, I fixed all the other errors , chucked unused things, and added to “bool useSerial.”

LMK how you’d like for me to handle the functor bit of code, and we’ll go from there.

All the best,

AJP
From: brian-kelley ***@***.***>
Reply-To: kokkos/kokkos-kernels ***@***.***>
Date: Thursday, September 16, 2021 at 12:35 PM
To: kokkos/kokkos-kernels ***@***.***>
Cc: Amy Jo Powell ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [kokkos/kokkos-kernels] Git submodules rps pr (#1037)


@brian-kelley requested changes on this pull request.

________________________________

In perf_test/blas/blas1/KokkosBlas_team_dot_perf_test.hpp<https://github.com/kokkos/kokkos-kernels/pull/1037#discussion_r710375984>:

> +  }

+};

+

+// Instantiate the machinery needed to run as an RPS test

+

+#ifdef KOKKOSKERNELS_ENABLE_TESTS_AND_PERFSUITE

+test_list construct_team_dot_kernel_base(const rajaperf::RunParams& params);

+#endif

+

+// Templated function

+template <typename ExecSpace, typename Layout>

+testData_rps_team_dot<ExecSpace, Layout> setup_test(int m, int repeat,

+                                                    bool layoutLeft,

+                                                    const int numberOfTeams);

+

+// Must have the full function body, as templated functions are recipes for

Remove notes

________________________________

In perf_test/blas/blas1/KokkosBlas_team_dot_perf_test.hpp<https://github.com/kokkos/kokkos-kernels/pull/1037#discussion_r710376415>:

> +      << ExecSpace::name() << ")\n";

+

+  std::cout << "Each test input vector has a length of " << m << std::endl;

+

+  // Declaring variable pool w/ a seeded random number;

+  // a parallel random number generator, so you

+  // won't get the same number with a given seed each time

+

+  // Constructing an instance of testData, which takes m as a param

+  testData_rps_team_dot<ExecSpace, Layout> testMatrices(m);

+

+  // do a warm up run of team dot:

+  Kokkos::parallel_for(

+      "TeamDotUsage_RPS_Warm_Up",

+      policy(testMatrices.numberOfTeams, Kokkos::AUTO),

+      KOKKOS_LAMBDA(

Use the functor you wrote in KokkosBlas_team_dot_perf_test.cpp instead of lambda, so that this will compile for CUDA without --expt-extended-lambda.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/kokkos/kokkos-kernels/pull/1037#pullrequestreview-756643889>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALV25GI6RZSMMU2AUG3MXB3UCI2GBANCNFSM475QBQPQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

ajpowelsnl commented:
@brian-kelley, @DavidPoliakoff , @lucbv  -- Closing this PR; required changes will be merged into develop via PR#1085.
